packages/angular-material/test/boolean-control.spec.ts

```diff
@@ -96,23 +160,181 @@ describe('Boolean control', () => {
-    }));
+        expect(checkboxInstance.checked).toBe(false);
+    });
+    it('should update with undefined value', () => {
+        const mockSubStore = MockNgRedux.getSelectorStub();
+        component.uischema = uischema;
 
-    it('should display errors', async(() => {
+        mockSubStore.next({
+            jsonforms: {
+                core: {
+                    data,
+                    schema,
+                }
+            }
+        });
+        fixture.detectChanges();
+        component.ngOnInit();
+        expect(component.data).toBe(true);
+        expect(checkboxInstance.checked).toBe(true);
+
+        mockSubStore.next({
+            jsonforms: {
+                core: {
+                    data: { foo: undefined },
+                    schema,
+                }
+            }
+        });
+        mockSubStore.complete();
+        fixture.detectChanges();
+        expect(component.data).toBe(undefined);
+        expect(checkboxInstance.checked).toBe(false);
+    });
+    it('should update with null value', () => {
+        const mockSubStore = MockNgRedux.getSelectorStub();
+        component.uischema = uischema;
+
+        mockSubStore.next({
+            jsonforms: {
+                core: {
+                    data,
+                    schema,
+                }
+            }
+        });
+        fixture.detectChanges();
+        component.ngOnInit();
+        expect(component.data).toBe(true);
+        expect(checkboxInstance.checked).toBe(true);
+
+        mockSubStore.next({
+            jsonforms: {
+                core: {
+                    data: { foo: null },
+                    schema,
+                }
+            }
+        });
+        mockSubStore.complete();
+        fixture.detectChanges();
+        expect(component.data).toBe(null);
+        expect(checkboxInstance.checked).toBe(false);
+    });
+    it('should not update with wrong ref', () => {
+        const mockSubStore = MockNgRedux.getSelectorStub();
+        component.uischema = uischema;
+
+        mockSubStore.next({
+            jsonforms: {
+                core: {
+                    data,
+                    schema,
+                }
+            }
+        });
+        fixture.detectChanges();
+        component.ngOnInit();
+        expect(component.data).toBe(true);
+        expect(checkboxInstance.checked).toBe(true);
+
+        mockSubStore.next({
+            jsonforms: {
+                core: {
+                    data: { foo: true, bar: false },
+                    schema,
+                }
+            }
+        });
+        mockSubStore.complete();
+        fixture.detectChanges();
+        expect(component.data).toBe(true);
+        expect(checkboxInstance.checked).toBe(true);
+    });
+    it('can be disabled', () => {
+        const mockSubStore = MockNgRedux.getSelectorStub();
+        component.uischema = uischema;
+        component.disabled = true;
+
+        mockSubStore.next({
```

Comment:
We don't think we need the store at all in this test case.

Reply:
probably not, I will recheck